frontend/app/src/App.tsx

```diff
@@ -605,10 +605,22 @@ export class App extends PureComponent<Props, State> {
     )
 
     if (newState === ConnectionState.CONNECTED) {
-      logMessage("Reconnected to server; requesting a script run")
-      // Trigger a full app rerun:
-      this.widgetMgr.sendUpdateWidgetsMessage(undefined)
-      this.setState({ dialog: null })
+      logMessage("Reconnected to server.")
+
+      const lastRunWasInterrupted =
```

Comment:
is this considered to be interrupted because otherwise we would see `ScriptRunState.STOPPED`?

Reply:
There's also `ScriptRunState.STOP_REQUESTED` and `ScriptRunState.COMPILATION_ERROR`, but yeah all of those states are those where we don't want to trigger another script run because they're either terminal states or moving toward them.